sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py

```diff
@@ -55,6 +87,19 @@ class EventHubProducerClient(ClientBase):
     :keyword str connection_verify: Path to the custom CA_BUNDLE file of the SSL certificate which is used to
      authenticate the identity of the connection endpoint.
      Default is None in which case `certifi.where()` will be used.
+    :keyword bool enable_idempotent_partitions: Indicates whether or not the producer should enable idempotent
+     publishing to the Event Hub partitions. If enabled, the producer will only be able to publish directly
+     to partitions; it will not be able to publish to the Event Hubs gateway for automatic partition routing
+     nor using a partition key. Default is False.
+    :keyword partition_configs: The set of configurations that can be specified to influence publishing behavior
+     specific to the configured Event Hub partition. These configurations are not necessary in the majority of
+     scenarios and are intended for use with specialized scenarios, such as when recovering the state used for
+     idempotent publishing.
+     It is highly recommended that these configurations only be specified if there is a proven need to do so;
+     Incorrectly configuring these values may result in an `EventHubProducerClient` instance that is unable to
+     publish to the Event Hubs. These configurations are ignored when publishing to the Event Hubs gateway for
+     automatic routing or when using a partition key.
```

Comment:
This might be self-explanatory (and not needed b/c, as you mentioned, our users don't need to be spoon-fed), but adding something at the end like the following could provide clarity:

If a dict is provided, instead of a PartitionPublishingConfiguration object, it may contain the optional parameters `producer_group_id`, `owner_level`, and `starting_sequence_number`.

Reply:
updated, thanks!